src/ui/statusbar.c

```diff
@@ -364,44 +383,49 @@ _extended_new(void)
 }
 
 static int
-_status_bar_draw_extended_tabs(int pos)
+_status_bar_draw_extended_tabs(int pos, gboolean prefix, int start, int end)
 {
     gint max_tabs = prefs_get_statusbartabs();
     if (max_tabs == 0) {
         return pos;
     }
 
-    if (g_hash_table_size(statusbar->tabs) > max_tabs) {
-        gboolean is_current = statusbar->current_tab > max_tabs;
+    guint opened_tabs = g_hash_table_size(statusbar->tabs);
+    if (g_hash_table_size(statusbar->tabs) <= max_tabs) {
+        return pos;
+    }
+
+    if (prefix && start < 2) {
+        return pos;
+    }
+    if (!prefix && end > opened_tabs - 1) {
+        return pos;
+    }
+    gboolean is_current = FALSE;
 
-        pos = _status_bar_draw_bracket(is_current, pos, "[");
+    pos = _status_bar_draw_bracket(is_current, pos, "[");
 
-        int status_attrs;
-        if (is_current) {
```

Comment:
actually i liked those 3 comments to easily see whats what. in the new code its harder to find this.

Reply:
sorry, what comments?

Reply:
anyway, there are only 2 states: new messages to the side, e.g. **[<]** or no new messages there `[<]` and `all other` comment doesn't make much sense and certainly doesn't improve code clarity.

Reply:
Hopefully now the comments are satisfying.